### DISCUSSION TITLE
Controllers & xinput/dinput Library overrides

### DISCUSSION THREAD

**Discussion by vvuk** [Whisky-App/Whisky#833] [General]
I just spent a while trying to figure out why my Xbox One Series X controller connected via bluetooth was not recognized in Elite Dangerous (or other games) under Whisky. Turns out the fix is simple -- in Wine configuration, Libraries, `xinput9_1_0` and `dinput8` need to be set to load the builtin library, not the native one (I have `builtin, native` configured).

I'm guessing this would be a common thing for lots of games... perhaps worth adding a checkbox directly in Whisky configuration to make this change?

**Comment by govind-maheshwari2** [comment]
Hi @vvuk can you share the steps or video to use builtin library.

**Reply by vvuk** [reply]
1. Select Bottle configuration
2. Press Wine Configuration button at the bottom
3. Select Libraries tab
4. Click on the libraries and press Edit. Select "builtin then native"
5. You may need to add some libraries like `dinput8` manually using the dropdown.
